### ISSUE TITLE
`CLIEngine#getFormatter()` is no longer a static function- throws an error

### ISSUE THREAD

**Issue by cowchimp** [eslint/eslint#5378]
`CLIEngine.getFormatter` used to be available as a static function.
It's also mentioned in [the docs](http://eslint.org/docs/developer-guide/nodejs-api#getformatter) that it's "available as a static function".
Since the introduction of the CLIEngine `cwd` option this is no longer the case since it calls `this.options.cwd` internally and `options` is undefined (because `options` is set with `defaultOptions` in the constructor which isn't called).

My use-case that made me encounter this is trying to use a custom formatter. I use `grunt-eslint` which calls `getFormatter` as static internally and passes a string as the argument, which means I expect it to infer it as a relative path which needs to concatenated to the current working directory.
Previously the cwd was resolved by calling `process.cwd()` and now it's resolved by calling `this.options.cwd` which as I mentioned is now undefined.

* **The version of ESLint you are using** 2.2.0
* **What you did** Call `getFormatter` as static and pass a path of a custom formatter as the argument
* **What you expected to happen** No error thrown. A custom formatter is returned.
* **What actually happened** Error: `Cannot read property 'cwd' of undefined`
* **The source code line that's causing the error** [here](https://github.com/eslint/eslint/blob/2bd57b6bbd11b23fdec65dc11c691f74c6a8db75/lib/cli-engine.js#L407)

**My configuration**

ESLint
```
{
    "ecmaFeatures": {
    },
    "rules": {
        "no-func-assign": 2,
        "no-delete-var": 2,
        "no-implicit-globals": 2
    }
}
```

grunt-eslint
```
eslint: {
	teamcity: {
		src: ['blabla/**/*.js'],
		options: {
			format: 'node_modules/eslint-teamcity/index.js'
		}
	}
}
```

**Output**

Running "eslint:teamcity" (eslint) task
Verifying property eslint.teamcity exists in config...OK
Files: blabla/yetayeta.js
Options: outputFile=false, quiet=false, maxWarnings=-1, format="node_modules/eslint-teamcity/index.js"
Warning: Cannot read property 'cwd' of undefined Use --force to continue.

**References**

* This is the issue that added the CLIEngine `cwd` option #4472
* This is the issue and pull request that made `getFormatter` static initially #3239.
* This is the issue I initially opened at `grunt-eslint` which uses `getFormatter` as a static method sindresorhus/grunt-eslint#109

Thank you!

**Comment by ilyavolodin** [comment]
This is a really good point. I didn't consider nature of `getFormatter` when I added `cwd` option. We should probably enhance it to use default value when `this.options` is not available.

**Comment by nzakas** [comment]
Agreed. @cowchimp, do you want to submit a PR to fix this?

**Comment by platinumazure** [comment]
@nzakas @ilyavolodin What's the value in having the static method in the first place, anyway? Do we have common use cases where people already have ESLint results and need a formatter and don't want to construct a new CLIEngine instance? In my own experimentation I've always just used CLIEngine all the way through, so I guess I've just never seen the need for the static method.

**Comment by cowchimp** [comment]
> Agreed. @cowchimp, do you want to submit a PR to fix this?

@nzakas Sure. Would be happy to we're if settled on the proposed change
(right now it looks like @ilyavolodin's suggestion to use `process.cwd()` as fallback).

> What's the value in having the static method in the first place, anyway? Do we have common use cases where people already have ESLint results and need a formatter and don't want to construct a new CLIEngine instance?

@platinumazure Would be interested in hearing @sindresorhus's opinion about this as the person who opened the issue for both exposing `getFormatter` as static and making `cwd` configurable (#3239 & #4472).

**Comment by sindresorhus** [comment]
I use it like this: https://github.com/sindresorhus/xo/blob/e9fb15e044601ee44b0b55fca0e588e8f2ecfaae/cli.js#L81

**Comment by nzakas** [comment]
@platinumazure these sort of questions are better answered by searching the issues/commit history for where the function came from. It was requested, and we added it. It was only the CWD change that made things hairy.

@cowchimp yup, I think @ilyavolodin's approach is correct.

**Comment by cowchimp** [comment]
OK, great. I'll try to get it done during the weekend. Thanks.

**Comment by cowchimp** [comment]
Hi. Please review my PR #5416.
LMK if you need me to change anything.
